### PR TITLE
ci: proper setting oldest/latest for integrations & `PL>=1.9`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Integrations on GPU were strangely set always oldest req.

Dropped compatibility with:
- `1.6` -  no special reason, but the next two versions wont work so skip also this old one
- `1.7` - Pl was looking for removed protected function, `ERROR integrations/test_lightning.py - ImportError: cannot import name '_compare_version' from 'torchmetrics.utilities.imports'`
- `1.8` - not aligned dependencies:
	```
	pytorch-lightning 1.8.0 depends on lightning-utilities==0.3.*
    torchmetrics 1.3.0.dev0 depends on lightning-utilities<0.10.0 and >=0.8.0
	```

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2129.org.readthedocs.build/en/2129/

### CODE REVIEW

**Comment by Borda** [comment]
@JustinGoheen @SkafteNicki seems the logging is different than expected for the latest PL (we have not tested it for long time :crying_cat_face: ) mind have a look, pls: https://dev.azure.com/Lightning-AI/Metrics/_build/results?buildId=176883&view=logs&j=12a9e589-df1e-541c-f0d0-67a8d22922d8&t=b6423d8e-d265-5686-c4f1-df41b5bef40a&l=279

**Comment by SkafteNicki** [comment]
cannot reproduce the error locally :/

**Comment by Borda** [comment]
> cannot reproduce the error locally :/

@awaelchli @justusschock any idea?

**Comment by awaelchli** [comment]
@Borda On master, if you check the [TM integration latest workflow](https://dev.azure.com/Lightning-AI/Metrics/_build/results?buildId=178275&view=logs&j=12a9e589-df1e-541c-f0d0-67a8d22922d8&t=25fd46e3-2f2b-58ce-2baf-d98fb6fb7915) you'll see that it installs pytorch-lightning 1.6.0, but this PR installs 2.0.9post. This is a pretty big jump. I suggest to do it in a smaller increment so that we can narrow it down easier.

File: src/torchmetrics/__init__.py [hunk 1]
```diff
@@ -11,6 +13,12 @@
 _PACKAGE_ROOT = os.path.dirname(__file__)
 _PROJECT_ROOT = os.path.dirname(_PACKAGE_ROOT)
 
+if package_available("PIL"):
```
Comment (Borda) [inline src/torchmetrics/__init__.py:16]: this is patch for the old PIL versions

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**